### DISCUSSION TITLE
Config flow could not be loaded: {"message":"Invalid handler specified"}

### DISCUSSION THREAD

**Discussion by mvinceti** [wlcrs/huawei_solar#276] [Setting up the integration]
Hi this is the error i get:

Logger: homeassistant.config_entries
Source: config_entries.py:852
First occurred: 19:38:28 (1 occurrences)
Last logged: 19:38:28

Error occurred loading configuration flow for integration huawei_solar: cannot import name 'AsyncModbusSerialClient' from 'pymodbus.client' (/usr/local/lib/python3.10/site-packages/pymodbus/client/__init__.py)

cant even pass first  screen when i wanna add integration that the mesage:

Config flow could not be loaded: {"message":"Invalid handler specified"}

**Comment by wlcrs** [comment]
cfr. #264

**Reply by mvinceti** [reply]
OK i downgrade to 1.6 and no more "invalid handler message"but  still cant connect there is error , i tried version 1.2. and  1.2.1  every time cant connect.

Now  only error "connection failed"
when download log file from inverter i can see that there is connection attempt from HA.

tried ethernet RTU and Wi-fi ,tried port 502 and 6607 and diferent slave ID (0, 1 ,2 or 3).


i must say that till this year, last December conection from HA and inverter worked fine over ethernet RTU or over WIFI,readings was ok and logging in Energy dash was correct.


Also i have MODBUS connection to one PLC (ethernet RU over port 502 ) and those reading are ok ( i had to made some adjutment 2 weeks ago) but now its is ok.

by the way very much thank You for Your effort
